### DISCUSSION TITLE
Supabase-js 2.0 select function is returning genericStringError

### DISCUSSION THREAD

**Discussion by devlukedreamstage** [supabase/supabase#8957] [Questions]
Attempting to call this function:

```
const { data, error } = await supabase
      .from("show")
      .select(attributes)
      .eq("slug", slug);
    if (error) throw error;
    return data;
```
Passing the attributes prop (a string which tells which attributes I want to fetch from the table) changes the promise to return a genericStringError

`Conversion of type 'GenericStringError | undefined' to type `

**Comment by cohlar** [comment]
Same here.
I have [generated types from the database](https://supabase.com/docs/reference/javascript/typescript-support) and injected into the supabase client, but the returned data remains of type `GenericStringError[] | null`.
